### ISSUE TITLE
Problem with webpart deployment to display form

### ISSUE THREAD

**Issue by milleraa** [SubPointSolutions/spmeta2#891]
Here is the model :

```cs
var webModel = SPMeta2Model
                            .NewWebModel(web =>
                            {
                                web
                                    .AddList(Lists.TestList1, list =>
                                    {
                                        list
                                            .AddHideContentTypeLinks(new HideContentTypeLinksDefinition
                                            {
                                                ContentTypes = { new ContentTypeLinkValue() { ContentTypeName = "Item" }, new ContentTypeLinkValue() { ContentTypeName = "Элемент" } }
                                            })
                                            .AddContentTypeLink(ContentTypes.MyType1)
                                            .AddListView(Views.View1)
                                            .AddHostListView(BuiltInListViewDefinitions.Lists.DispForm, view =>
                                            {
                                                view.AddWebPart(WebParts.TestWebPart3);
                                            });
                                    })
                                    .AddList(Lists.TestList2, list =>
                                    {
                                        list
                                            .AddHideContentTypeLinks(new HideContentTypeLinksDefinition
                                            {
                                                ContentTypes = { new ContentTypeLinkValue() { ContentTypeName = "Item" }, new ContentTypeLinkValue() { ContentTypeName = "Элемент" } }
                                            })
                                            .AddContentTypeLink(ContentTypes.MyType2)
                                            .AddListView(Views.View2)
                                            .AddHostListView(BuiltInListViewDefinitions.Lists.AllItems, view =>
                                             {
                                                 view.AddWebPart(WebParts.TestWebPart1);
                                             });
                                    })
                                    .AddList(Lists.TestList3, list =>
                                    {
                                        list
                                            .AddHideContentTypeLinks(new HideContentTypeLinksDefinition
                                            {
                                                ContentTypes = { new ContentTypeLinkValue() { ContentTypeName = "Item" }, new ContentTypeLinkValue() { ContentTypeName = "Элемент" } }
                                            })
                                            .AddContentTypeLink(ContentTypes.MyType3)
                                            .AddListView(Views.View3)
                                            .AddHostListView(BuiltInListViewDefinitions.Lists.DispForm, view =>
                                            {
                                                view.AddWebPart(WebParts.TestWebPart2);
                                            });
                                    });


                            });
```
Expected result:
1. DispForm in List1 contains WebPart3
2. AllItems.aspx in List2 contains WebPart1
3. DispForm in List3 contains WebPart2

Actual result:
1. DispForm in List1 contains WebPart2 and WebPart3
2. Allitems.aspx in List2 contains WebPart1
3 DispForm in List3 contains nothing but standart ListFormWebPart.

**Comment by milleraa** [comment]
```
PS C:\Users\spfa> $f.BuildVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
15     0      4841   1000
```

**Comment by SubPointSupport** [comment]
Confirming the issue. Investigating.

**Comment by SubPointSupport** [comment]
Fixed, running tests on list views regression and will push to dev. Will be in the 1.2.90 release on Monday.

Seems that if you have a XsltListViewWebPart on list forms-pages.
The result targetList.Views collection would have ALL list views + your binded view from the web part. Here is an output from the debug:
```
// 	targetList.RootFolder.Url	"Lists/TestList3"	string
//  targetList.Views[0].ServerRelativeUrl	"/Lists/TestList3/AllItems.aspx"	string
//  targetList.Views[1].ServerRelativeUrl	"/Lists/TestList1/DispForm.aspx"	string
//  targetList.Views[2].ServerRelativeUrl	"/Lists/TestList3/AllItems3.aspx"	string
```

Seems to be a pure SharePoint API issue. We added additional filtering so that the target page / file is found right. The result on your model looks good, exactly as expected.
